### DISCUSSION TITLE
Axis Movement - Absolute or Relative?

### DISCUSSION THREAD

**Discussion by Wombat37** [bigtreetech/BIGTREETECH-TouchScreenFirmware#2219] [Ideas]
When I want to manually move the extruder head using the TFT, it's normally because I want to move it out of the way in an emergency or to a position where I can service it.

Usually I get a message telling me that the head can't be moved because it hasn't been homed - this rather renders the function useless to me. Homing in an emergency is not an option and if I just want to raise the head. Having to home it first is really tedious and not usually possible.

Is there any reason that the code can't be changed to make relative movements (i.e. from where the head is at the moment)?  

Obviously for macros etc., absolute positioning is a must, but for manual movements from the TFT, I would have thought that relative movements would be more useful because you're usually moving the head from where it currently is (up, down, left, right, forwards and backwards)..

Is this a reasonable suggestion or am I missing something here?

**Comment by kisslorand** [comment]
Sounds to me like a Marlin config issue rather than the TFT FW.
BTW,  better move this to the "Issues" section, it's more likely to have your questions answered. There's a button on the right for this labeled "Create issue from discussion".

**Reply by Wombat37** [reply]
Why would Marlin effect what the TFT controls do on the movement page? Surely it's what the TFT sends to Marlin that's important here?

We had a similar issue with the extruder movement earlier - the TFT control applied absolute rather than relative settings (and dumped meters of filament out through the extruder).

I'm using RRF anyway.

**Reply by kisslorand** [reply]
> Why would Marlin effect what the TFT controls do on the movement page? 

There are two major reasons:

1.  There is no such configuration on the TFT FW and for other people (including myself) it just works fine
2.  In Marlin there's a setting for this (`#define NO_MOTION_BEFORE_HOMING`)

**Comment by Wombat37** [comment]
OK - I stand corrected and thank you for the clarification.
